src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -717,9 +718,11 @@ protected override BoundNode VisitReturnStatementNoAdjust(BoundReturnStatement n
             TypeSymbolWithAnnotations returnType = GetReturnType(compilation, _currentMethodOrLambda);
             if ((object)returnType != null)
             {
-                TypeSymbolWithAnnotations unconvertedType = result.Type;
-                Conversion conversion = GenerateConversion(_conversions, expr, unconvertedType?.TypeSymbol, returnType.TypeSymbol);
-                TypeSymbolWithAnnotations resultType = InferResultNullability(expr, conversion, returnType.TypeSymbol, unconvertedType);
+                TypeSymbolWithAnnotations resultType = ApplyConversion(expr, expr, conversion, returnType.TypeSymbol, result.Type, checkConversion: true, fromExplicitCast: false, out bool canConvert);
+                if (conversion.Exists && !canConvert)
```

Comment:
>canConvert [](start = 42, length = 10)

The name is somewhat confusing given that we checked that ```conversion.Exists```. Consider renaming to "topLevelNullablilityMatches", if that accurately reflects the meaning of the local.

Reply:
Renamed to `canConvertNestedNullability`.

---
In reply to: [190722193](https://github.com/dotnet/roslyn/pull/26356#discussion_r190722193) [](ancestors = 190722193)